app/main/lib/shared_models/video_model.py

```diff
@@ -137,14 +153,18 @@ def search(self, task):
                 if l1_scores[i] > app.config['VIDEO_MODEL_L1_SCORE']:
                     qualified_matches.append(match)
             files = self.get_fullpath_files(qualified_matches, False)
+            if self.tmk_file_exists(video.folder, video.filepath):
+                folder, filepath = (video.folder, video.filepath)
+            else:
+                folder, filepath = self.download_temp_file(task)
```

Comment:
I'm concerned about this call @DGaffney . I think we should be passing in `result` and not `task` to this function. Unless there is a side-effect somewhere that I'm not seeing, `task` exists before we call Presto and does not have the location we actually want to download from.

Instead, what appears to be happening is that `download_temp_file` makes another call to `get_blocked_response`, which calls Presto another time. Can you check the logs locally and ensure Presto is only being called once for one innovation of this API endpoint on Alegre?

Reply:
I'm actually not sure about that @computermacgyver - the first call to Presto where `if video.hash_value is None:` is *never* true for the cases that we are treating - at the time of evaluation, the video exists, and it has a valid `hash_value`. Thus, we need a different check for if the file is missing. If you want, we could conflate both cases so they go through the same routine (e.g. `if video.hash_value is None or not self.tmk_file_exists(video.folder, video.filepath):`), but we'd still need to do the `if..else` block within that check, as we don't want to download a temporary version of the fingerprint in case its valid and already on disk:

```
if self.tmk_file_exists(video.folder, video.filepath):
    folder, filepath = (video.folder, video.filepath)
else:
    folder, filepath = self.download_temp_file(task)
```

If you want to ensure that we can only ever call `get_blocked_response` once for any possible iteration/combination of state, we can certainly refactor to accommodate that, but this particular issue is distinct from that existing `get_blocked_response` call, which handles a (iirc) much rarer edge case (and imo maybe even should be removed at this point)